### DISCUSSION TITLE
Micropython on ESP32-C3 GPIO18 and GPIO19 cannot be configured as ordinary GPIO

### DISCUSSION THREAD

**Discussion by orztrickster** [micropython/micropython#13168] [ESP32] [answered]
`from machine import Pin`
`gpio18=Pin(18, Pin.OUT)`
ValueError: invalid pin

How do I set this up?

**Comment by robert-hh** [comment, marked as answer]
Pin 18 and 19 are reserved for Serial JTAG when the respective flag is set. That flag is set for ESP32_GENERIC_C3 and LOLIN_C3_MINI. To get rid of that, you would have to build your own firmware and remove the setting of `CONFIG_ESP_CONSOLE_USB_SERIAL_JTAG=y` in `boards/ESP32_GENERIC_C3/sdkconfig.c3usb`.

**Reply by orztrickster** [reply]
Thank you for your response.
Is it necessary to modify the sdkconfig.c3usb file? Is there a way to directly modify it within the MicroPython code?

**Reply by robert-hh** [reply]
You can of course get it to work using Python code by programming the respective registers of the GPIO using machine.mem32() methods. But that is tedious.

**Reply by JustDr00py** [reply]
The new micropython doesnt have this option? any suggestions.
